docs/source/notes/windows.rst

```diff
@@ -0,0 +1,256 @@
+
+.. code-block:: bat
+
+    conda install -c peterjc123 vc vs2017_runtime
+
+Another possible cause may be you are using GPU version without NVIDIA
+graphics cards. Please replace your GPU package with the CPU one.
+
+Usage (multiprocessing)
+-------------------------------------------------------
+
+Multiprocessing error without if-clause protection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: py3tb
+
+    RuntimeError:
+   	An attempt has been made to start a new process before the
+   	current process has finished its bootstrapping phase.
+
+       This probably means that you are not using fork to start your
+       child processes and you have forgotten to use the proper idiom
+       in the main module:
+
+           if __name__ == '__main__':
+               freeze_support()
+               ...
+
+       The "freeze_support()" line can be omitted if the program
+       is not going to be frozen to produce an executable.
+
+The implementation of ``multiprocessing`` is different on Windows, which
+uses ``spawn`` instead of ``fork``. So we have to wrap the code with an
+if-clause to protect the code from executing multiple times. Refactor
+your code into the following structure.
+
+.. code-block:: python
+
+    import torch
+
+    def main()
+        for i, (x, y) in dataloader:
+            # do something here
+
+    if __name__ == '__main__':
+        main()
+
+
+Multiprocessing error "Broken pipe"
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: py3tb
+
+    ForkingPickler(file, protocol).dump(obj)
+
+    BrokenPipeError: [Errno 32] Broken pipe
+
+This issue happens when the child process ends before the parent process
+finishes sending data. There may be something wrong with your code. You
+can debug your code by reducing the ``num_worker`` of ``DataLoader`` to zero
+and see if the issue persists.
+
+Multiprocessing error "driver shut down"
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: py3tb
+
+    Couldn’t open shared file mapping: <torch_14808_1591070686>, error code: <1455> at torch\lib\TH\THAllocator.c:154
+
+    [windows] driver shut down
+
+Please update your graphics driver. If this persists, this may be that your
+graphics card is too old or the calculation is too heavy for your card. Please
+update the TDR settings according to this `post
+<https://www.pugetsystems.com/labs/hpc/Working-around-TDR-in-Windows-for-a-better-GPU-computing-experience-777/>`_.
+
+CUDA IPC operations
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: py3tb
+
+   THCudaCheck FAIL file=torch\csrc\generic\StorageSharing.cpp line=252 error=63 : OS call failed or operation not supported on this OS
+
+They are not supported on Windows. Something like doing multiprocessing on CUDA
+tensors cannot succeed, there are two alternatives for this.
+
+1. Don't use `multiprocessing`. Set the ``num_worker`` of ``DataLoader`` to zero.
+2. Share CPU tensors instead. Remove ``.cuda()`` in your custom DataLoader definition.
```

Comment:
nit1: It should be custom `Dataset` definition (or better ``:class:`~torch.utils.data.Dataset` ``).

nit: Can you also mention the similar effect `.to()` calls? Or maybe we should just say that the dataset should return CPU tensors.